frontend/libs/studio-pure-functions/src/TextResourceUtils/TextResourceUtils.ts

```diff
@@ -53,6 +53,29 @@ export class TextResourceUtils {
     return new TextResourceUtils(newMap);
   }
 
+  public setMultiple(textResources: TextResource[]): TextResourceUtils {
+    const newMap = this.cloneMap();
+    textResources.forEach((textResource) => {
+      newMap.set(textResource.id, textResource);
```

Comment:
We could use `TextResourceUtils.set` here, but that would create a new copy of the list on each iteration. It's not necessary to do that more than once.